### PR TITLE
PLP + Search using Live Search

### LINKED ISSUE #10: Product List Page
- [x] Depends on https://experienceleague.adobe.com/docs/commerce-merchant-services/live-search/overview.html?lang=en and availability of monolith GraphQL endpoint
- [x] Investigate if Reflektion API from #76 is suitable for PLP
- [ ] Implement Product List Page block
    - [x] Get product data from live search.
    - [ ] ~~Get category data from monolith GraphQL endpoint~~
    - [x] With facets
    - [x] Variant selection for products
    - [x] Sorting
    - [x] Pagination + Number of items per page

### LINKED ISSUE #76: Product Search
### Phase 1
- [x] Setup search popup via Sitecore / Reflektion
- [x] ~~Create Product Search Page block and implement via Sitecore / Reflektion~~
### Phase 2
- [x] ~~Setup LiveSearch according to https://experienceleague.adobe.com/docs/commerce-merchant-services/live-search/onboard/install.html?lang=en~~
- [x] Make Reflektion API available
    - via https://secretshare.corp.adobe.com/?42e5733a1838a77c&k=DPrwyiQyQxF2ffYzpawPXJLYpPuEkVU49TQUN1LX6pkM
- [ ] Implement Product Search Page block, re-use PLP from https://github.com/hlxsites/maidenform/issues/10
- [x] Use Reflektion API to fill block with data

### PR DESCRIPTION
**Pull request by herzog31**
Please always provide the [GitHub issue(s)](../issues) your PR is for, as well as test URLs where your change can be observed (before and after):

Fix #10
Fix #76

Test URLs:
- Before: https://main--maidenform--hlxsites.hlx.page/
- After:
    - Search: https://issue-76-ls--maidenform--hlxsites.hlx.live/search?q=comfy
    - PLP: https://issue-76-ls--maidenform--hlxsites.hlx.live/bras
    - Homepage: https://issue-76-ls--maidenform--hlxsites.hlx.live
    - PDP: https://issue-76-ls--maidenform--hlxsites.hlx.live/products/everyday-full-coverage-cushioned-underwire-bra/09436

### CODE REVIEW

File: blocks/product-list-page/productSearch.graphql.js [hunk 1]
```diff
@@ -0,0 +1,89 @@
+const query = `query productSearch(
```
Comment (dicagno) [inline blocks/product-list-page/productSearch.graphql.js:1]: @herzog31 do you think we might use some gql query builder for cases like this in the future? maybe for making complex queries less error-prone

**Review by dicagno — APPROVED** [review]
LGTM, great job!

File: blocks/product-list-page/productSearch.graphql.js (hunk 1 above)
Comment (herzog31) [reply blocks/product-list-page/productSearch.graphql.js:1]: Not during runtime, but maybe having something during build time (once we have a build step) that checks the query would be great.

File: scripts/commerce.js [hunk 2]
```diff
@@ -0,0 +1,319 @@
+/* eslint-disable import/prefer-default-export */
```
[the remark this answers is not in the document]
Comment (herzog31) [reply scripts/commerce.js:1]: Merged both into `commerce.js`

File: blocks/product-list-page/ProductList.js [hunk 3]
```diff
@@ -51,14 +57,91 @@ class ProductCard extends Component {
         this.resizeObserver.disconnect();
       }
     }
+  }
+
+  renderPrice(product) {
+    // Simple product
```
[the remark this answers is not in the document]
Comment (herzog31) [reply blocks/product-list-page/ProductList.js]: I created a new `renderPrice` method in `commerce.js` which is used by both blocks.

**Pull request merged**